redis/Cargo.toml

```diff
@@ -76,11 +84,16 @@ geospatial = []
 json = ["serde", "serde/derive", "serde_json"]
 cluster = ["crc16", "rand"]
 script = ["sha1_smol"]
-tls = ["native-tls"]
+tls-native-tls = ["native-tls"]
```

Comment:
Can we not rename the `tls` feature?

Reply:
Is this the only breaking change?

Reply:
This, along with the async-std + native-tls feature, are the only breaking changes.

I renamed them simply because the naming convention wouldn't be consistent with Rustls. Also, `async-std-tls-comp` and `tokio-native-tls-comp` are two features with inconsistent names (since both depend on `native-tls`). Further, using a feature called `tls` implies a default TLS implementation, which IMHO should be left up to the user (since both have their pros and cons).

Reply:
Your points are totally reasonable. I'm just a bit weary of breaking users unnecessarily just for the name change. Could we potentially "deprecate" the features, e.g., change `tls` to
```
tls = ["tls-native-tls"]
```

Likewise for `async-std-tls-comp`, add some comments nothing such, and we can remove them later?

Reply:
Okay, sure. I'll make the changes and add deprecation warnings. In which file would you prefer to add them? `lib.rs`?

Reply:
Thanks! Wherever we already mention TLS is fine (which offhand I think is just the README?) as well as above the "deprecated" features in Cargo.toml.

I don't think there is any actual way to deprecate a feature, but if there is, I guess we should do that too :laughing:.

Reply:
I added the `tls` and `async-std-tls-comp` features back, with a deprecation notice both in the README and in `Cargo.toml`.

As for adding compile-time deprecation notices, I don't think there's an official way (since https://github.com/rust-lang/cargo/issues/7130 exists), unless you want to manually add a compile-time warning with the help of a third-party crate (like [compile-warning](https://crates.io/crates/compile_warning)).